test-data/unit/check-literal.test

```diff
@@ -3093,3 +3093,13 @@ def test() -> None:
     if x == 2:  # OK
         ...
 [builtins fixtures/bool.pyi]
+
+[case testNegativeIntLiteral]
+from typing_extensions import Literal
+
+a: Literal[-2] = -2
+b: Literal[-1] = -1
+c: Literal[0] = 0
+d: Literal[1] = 1
+e: Literal[2] = 2
+[out]
```

Comment:
You need to add a test fixture here, probably patch `fixtures/float.pyi` by adding `__neq__` there, and use it here.